### ISSUE TITLE
enum validation

### ISSUE THREAD

**Issue by jezzsantos** [markusbohl/fluent-ts-validator#5]
given an enum like this:
```
export enum species {
    none = "",
    cattle = "cattle",
    deer = "deer"
}
```

and I have a DTO defined like this:
```
class AClass{
    species: species = species.none;
}
```

I would love a validator to make sure I have one of the values in the enum, something like this perhaps?
```
        this.validateIfString(dto => dto.species)
            .isOneOf(species)
            .whenNotEmpty()
```

`isOneOf(enum)` or `isIn(enum)` is that doable?

**Comment by markusbohl** [comment]
I don't know yet. But I'll figure out, if it's doable or not.

**Comment by markusbohl** [comment]
Okay. I've implemented a possible solution in a feature-branch as you can see from the commits above.
It would work like this:
```
enum StringColors {
    Red = "RED",
    Green = "GREEN",
    Blue = "BLUE"
}

enum NumColors {
    Black = 0,
    White = 1
}

class SomeColors {
    color1: StringColors;
    color2: NumColors;
    color3: string;
    color4: number;
}

class ColorValidator extends AbstractValidator<SomeColors> {
    constructor() {
        super();
        this.validateIf(dto => dto.color1) // type StringColors
            .isIn(StringColors);
        this.validateIf(dto => dto.color2) // type NumColors
            .isIn(NumColors);
        this.validateIf(dto => dto.color3) // type string
            .isIn(StringColors);
        this.validateIf(dto => dto.color4) // type number
            .isIn(NumColors);
    }
}

const someColors = new SomeColors();
someColors.color1 = StringColors.Blue;
someColors.color2 = NumColors.Black;
someColors.color3 = "GREEN";
someColors.color4 = 1;

new ColorValidator().validate(someColors); // valid
```

Since `enum` members can either be of type `number` or of type `string` (since TypeScript 2.4) I think it is useful to use the more general `validateIf()` method instead of `validateIfNumber()` or `validateIfString()`. This makes the validator more robust in case of type changes within the enum (i.e. switching from `number` to `string` initializers).

The `enum` fields as well as `string` or `number` fields of your object under validation can now be validated against the members of enums. Actually, this would not only work for the `isIn()` case but also for the opposite `isNotIn()` case.

What do you think? Would such an implementation meet your needs? Or do you think, we should do something different? Let me know what you think.

**Comment by jezzsantos** [comment]
Thanks for this @markusbohl I'll try it out shortly. 
But in terms of design as you point out, I am happy with which way you want to go.

**Comment by markusbohl** [comment]
I've published a release candidate `fluent-ts-validator@3.0.0-rc.1` on npm. This should make it a bit easier for you to test.

**Comment by markusbohl** [comment]
Just released version [3.0.0](https://github.com/markusbohl/fluent-ts-validator/releases/tag/3.0.0) with the solution as shown above.
